aw/ViewControllers/News/NewsTableViewController.swift

```diff
@@ -0,0 +1,193 @@
+import CoreData
+import UIKit
+
+class NewsTableViewController: UITableViewController {
+    @IBOutlet weak var donateButton: UIButton!
+
+    var managedObjectContext: NSManagedObjectContext?
+    var fetchedResultsController: NSFetchedResultsController<Article>?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        initialize()
+
+        if let lastUpdated = DefaultsManager.articlesLastUpdated {
+            if lastUpdated < Date(timeIntervalSinceNow: -86400) {
+                updateArticles()
+            }
+        } else {
+            updateArticles()
+        }
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        switch segue.identifier {
+        case Segue.articleDetail.rawValue:
+            guard let articleVC = segue.destination as? SingleArticleViewController,
+                let indexPath = tableView.indexPathForSelectedRow,
+                let article = fetchedResultsController?.object(at: convertToFetchedResults(indexPath)) else { return }
+            articleVC.article = article
+        default:
+            print("unknown segue")
+        }
+    }
+
+    // MARK: - Table view data source
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            return 1
+        default:
+            return fetchedResultsController?.fetchedObjects?.count ?? 0
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch indexPath.section {
+        case 0:
+            guard let cell = tableView.dequeueReusableCell(
+                withIdentifier: "supportAWCell",
+                for: indexPath) as? NewsDonateTableViewCell else {
+                fatalError("Failed to deque donate cell")
+            }
+            return cell
+        default:
+            guard let cell = tableView.dequeueReusableCell(
+                withIdentifier: "newsCell",
+                for: indexPath) as? NewsTableViewCell else {
+                fatalError("Failed to deque newsCell")
+            }
+
+            let article = fetchedResultsController?.object(at: convertToFetchedResults(indexPath))
+
+            cell.article = article
+            cell.update()
+            return cell
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        switch indexPath.section {
+        case 0:
+            return 204
```

Comment:
It might be clearer to have these values set as part of the cell views, rather than as magic numbers here. That will also be more modular when adding new types of cells.

Reply:
Do you use `UITableViewAutomaticDimension` then?

Reply:
Yeah, you probably shouldn't hard code the size at all. Just set up the appropriate constraints and let `UITableViewAutomaticDimension` size as appropriate.